### PR TITLE
JBEAP-26402: handle badly formatted artifact cache

### PR DESCRIPTION
**Pull request by michpetrov**
Issue: [JBEAP-26402](https://issues.redhat.com/browse/JBEAP-26402)

### CODE REVIEW

**Review by spyrkob — CHANGES_REQUESTED** [review]
Thanks @michpetrov. A couple of comments, plus could you port it to 1.1.x branch as well please?

File: prospero-common/src/main/java/org/wildfly/prospero/galleon/ArtifactCache.java [hunk 1]
```diff
@@ -205,6 +205,9 @@ private void init() throws IOException {
                 final List<String> lines = Files.readAllLines(artifactLog);
                 for (String line : lines) {
                     final String[] splitLine = line.split(ArtifactCache.CACHE_LINE_SEPARATOR);
+                    if (splitLine.length < 3) {
+                        throw new MavenUniverseException("Bad record format");
```
[the remark this answers is not in the document]
Comment (michpetrov) [reply prospero-common/src/main/java/org/wildfly/prospero/galleon/ArtifactCache.java:209]: If I add this to the logger what about the IOE in the catch block? I'd assume we want the top message to be the same, i.e. "can't read cache".

[the remark this answers is not in the document]
Comment (spyrkob) [reply prospero-common/src/main/java/org/wildfly/prospero/galleon/ArtifactCache.java:209]: ah good point, I remember why it was using MavenUniverseException. It's the top level catch that should have a more descriptive error message

[the remark this answers is not in the document]
Comment (michpetrov) [reply prospero-common/src/main/java/org/wildfly/prospero/galleon/ArtifactCache.java:209]: @spyrkob well what messages do you want me to change then?

[the remark this answers is not in the document]
Comment (spyrkob) [reply prospero-common/src/main/java/org/wildfly/prospero/galleon/ArtifactCache.java:209]: @michpetrov the one in `throw new IOException("Unable to read cached items.", e);` if you don't mind adding this to this PR

**Comment by michpetrov** [comment]
@spyrkob does this work? I'll port it to 1.1.x if there're no further requests.

**Review by spyrkob — APPROVED** [review]
Thanks @michpetrov that looks great!

**Pull request merged**